### ISSUE TITLE
Strong name sign assemblies

### ISSUE THREAD

**Issue by GieltjE** [mgp25/Chat-API-NET#208]
Please strong name sign the .dll files (including the May, mycurve25519 and mylibaxolotl files), this is very important for projects that sign their files (will result in a failure to load the unsigned files).

**Comment by jwjosefy** [comment]
Here's a tutorial on how to do it
http://bfy.tw/7Epx

Em ter, 16 de ago de 2016 08:41, Michiel Hazelhof <notifications@github.com>
escreveu:

**Comment by GieltjE** [comment]
I know how to do it, that is not the problem nor the point.

All mayor nuget packages deliver their packages presigned, probably for the following reasons:
1: If the signature suddenly changes someone might have compromised their account (our build software breaks in this case)
2: Makes redistribution a lot easier, if we pull the package again from nuget we would have to resign it every time

**Comment by jwjosefy** [comment]
Well, let me put it in perspective:
Do you realize that to be able to sign one needs a key, which is unique and
traceable back to the owner... And that's the whole discord point. If you
want to sign, you are free to create one key for yourself and do it. But
don't expect another person to sign it on your behalf.
I don't mean to be rude, only being realistic.

Em ter, 16 de ago de 2016 09:04, Michiel Hazelhof <notifications@github.com>
escreveu:

**Comment by GieltjE** [comment]
Being realistic is creating one and signing it, if one accepts a build and the next one has the same signature one doesn't have to do all the checks again.

This is the first package on nuget that I encountered which doesn't strong name sign the files.

Nuget was created to be able to get libraries with ease, having to resign them every time before distribtion doesn't quite fit that glove.

**Comment by jwjosefy** [comment]
You are right about Nuget. On the other hand, it was also conceived to host
opensource projects by heart, not necessarily projects like this one that
reverse engineers a closed source system. So, by definition, there's no
obligation to provide a signed assembly.

I'm certain you are very well aware of the ethical / legal implications of
using this library.

Em ter, 16 de ago de 2016 09:11, Michiel Hazelhof <notifications@github.com>
escreveu:

**Comment by GieltjE** [comment]
I am very aware and we have limited the usage to prevent abuse (and keep tight logs).

Although it is not a requirement for nuget packages it is one that I feel contributes to a better ecosystem.

If you still feel it is a no-go please just close this report
